packages/calling-stateful-client/src/CallClientState.ts

```diff
@@ -664,6 +664,29 @@ export interface AcceptedTransfer {
   timestamp: Date;
 }
 
+/**
+ * State to track the types {@link CallInfo} and {@link TeamsCallInfo}
+ * @public
+ */
+export interface CallInfoState {
+  /**
+   * GroupId of the call that you joined
+   */
+  groupId?: string;
+  /**
+   * The teams meeting thread id
+   */
+  threadId?: string;
+  /**
+   * participant id of the local user
+   */
+  participantId: string;
+  /**
+   * Differentiator between the Call and TeamsCall types
```

Comment:
Why did this get merged into a single CallInfoState and not CallInfoState and TeamsCallInfoState?
What if CallInfoState and TeamsCallInfoState respectively diverge (i.e. TeamsCallInfoState has properties that CallInfoState does not and vice versa), we won't be able to maintain this interface?

Reply:
Or when they diverge that's the point we add CallInfoState and TeamsCallInfoState?

Reply:
We decided to flatten it in since there is no differentiation between them. unless the TeamsCallInfo property gets its own properties at some point they are same except its missing the groupId property. So we could either keep adding to it directly, or as you say make it a common when that does happen. I don't believe @PorterNan and I discussed that outcome when we made this decision.